stl/inc/format

```diff
@@ -2557,8 +2557,221 @@ template <_Format_supported_charT _CharT, class _Traits>
+        return _Ptr;
+    }
+    const _T* end() const noexcept {
+        return _Ptr + _Size;
+    }
+
+    size_t size() const noexcept {
+        return _Size;
+    }
+
+    size_t capacity() const noexcept {
+        return _Capacity;
+    }
+
+    _T* data() noexcept {
+        return _Ptr;
+    }
+
+    const _T* data() const noexcept {
+        return _Ptr;
+    }
+
+    void clear() {
+        _Size = 0;
+    }
+
+    void try_resize(size_t _Count) {
+        try_reserve(_Count);
+        _Size = _Count <= _Capacity ? _Count : _Capacity;
+    }
+
+    void try_reserve(size_t _New_capacity) {
+        if (_New_capacity > _Capacity)
+            grow(_New_capacity);
+    }
+
+    void push_back(const _T& _Value) {
+        try_reserve(_Size + 1);
+        _Ptr[_Size++] = _Value;
+    }
+
+    template <class _U>
+    void append(const _U* _Begin, const _U* _End);
+
+    template <class _I>
+    _T& operator[](_I _Index) {
+        return _Ptr[_Index];
+    }
+    template <class _I>
+    const _T& operator[](_I _Index) const {
+        return _Ptr[_Index];
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(size_t) {}
+    size_t count() const {
+        return 0;
+    }
+    size_t limit(size_t _Size) {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    size_t _Count = 0;
+    size_t _Limit;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(size_t _L) : _Limit(_L) {}
+    size_t count() const {
+        return _Count;
+    }
+    size_t limit(size_t _Size) {
+        size_t _N = _Limit > _Count ? _Limit - _Count : 0;
+        _Count += _Size;
+        return _Size < _N ? _Size : _N;
+    }
+};
+
+template <class _OutputIt, class _T, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_T> {
+private:
+    _OutputIt _Out;
+    enum { _Buffer_size = 256 };
+    _T _Data[_Buffer_size];
+
+protected:
+    void grow(size_t) final override {
+        if (this->size() == _Buffer_size)
+            flush();
+    }
+    void flush();
+
+public:
+    explicit _Fmt_iterator_buffer(_OutputIt _O, size_t _N = _Buffer_size)
+        : _Traits(_N), _Fmt_buffer<_T>(_Data, 0, _Buffer_size), _Out(_O) {}
+    ~_Fmt_iterator_buffer() {
+        flush();
+    }
+
+    _OutputIt out() {
+        flush();
+        return _Out;
+    }
+    size_t count() const {
+        return _Traits::count() + this->size();
+    }
+};
+
+template <class _T>
+class _Fmt_iterator_buffer<_T*, _T> final : public _Fmt_buffer<_T> {
+protected:
+    void grow(size_t) final override {}
+
+public:
+    explicit _Fmt_iterator_buffer(_T* _Out, size_t = 0) : _Fmt_buffer<_T>(_Out, 0, ~size_t()) {}
+
+    _T* out() {
+        return &*this->end();
+    }
+};
+
+template <class _Container>
+class _Fmt_iterator_buffer<_STD back_insert_iterator<_Container>,
+    enable_if_t<_STD contiguous_iterator<typename _Container::iterator>, typename _Container::value_type>>

```

Comment:
I would defer to @StephanTLavavej or @CaseyCarter but conventionally we use a final `enable_if_t,< meow, int> = 0` for SFINAE

Reply:
Note that it's a class template.

Reply:
I don't think you can add template parameters to classes like this: https://godbolt.org/z/KExz9Ezb6.